### PR TITLE
Add tests for eth_getLogs

### PR DESCRIPTION
**Pull request by lyubenkov**

### CODE REVIEW

File: integration/tests/basic/test_rpc_calls.py [hunk 1]
```diff
@@ -101,6 +103,13 @@ def get_event_signatures(abi: tp.List[tp.Dict]) -> tp.List[str]:
 class TestRpcCalls(BaseMixin):
     _erc20_contract: tp.Optional[tp.Any] = None
 
+    @pytest.fixture(scope="class")
+    def event_caller_contract(self, web3_client, class_account) -> typing.Any:
```
Comment (kristinaNikolaevaa) [inline integration/tests/basic/test_rpc_calls.py]: Let's move it to conftest and delete duplication (for example test_event_logs.py )

File: integration/tests/basic/test_rpc_calls.py [hunk 2]
```diff
@@ -194,22 +203,47 @@ def test_eth_gas_price(self, param: tp.Union[str, None]):
...
-        contract, deploy_tx, _ = erc20_contract
-        topics = get_event_signatures(contract.abi)
-        params = {
-            "address": deploy_tx.contractAddress,
-            "topics": topics,
-        }
+    @pytest.mark.parametrize(
+        "param_fields", [("address", "topics"), ("address",), ("topics",)]
+    )
+    def test_get_rpc_logs(self, event_caller_contract, param_fields, tag1, tag2):
```
Comment (kristinaNikolaevaa) [inline integration/tests/basic/test_rpc_calls.py:302]: it is still  test to check logs via tags

check this page please
https://ethereum.org/en/developers/docs/apis/json-rpc/

fromBlock and toBlock can be tags and QUANTITY value

address, topics and blockhash - optional values
we should check all variants and negative tests too

**Review by kristinaNikolaevaa — CHANGES_REQUESTED** [review]

File: integration/tests/basic/test_rpc_calls.py (hunk 1 above)
Comment (lyubenkov) [reply integration/tests/basic/test_rpc_calls.py]: Good point, done!

File: integration/tests/basic/test_rpc_calls.py (hunk 2 above)
Comment (lyubenkov) [reply integration/tests/basic/test_rpc_calls.py:302]: Added tests for fromBlock&toBlock QUANTITY value, blockHash and negative scenarios too.

File: integration/tests/conftest.py [hunk 3]
```diff
@@ -235,3 +235,11 @@ def meta_proxy_contract(web3_client, faucet, class_account):
         "./EIPs/MetaProxy.sol", "0.8.10", account=class_account
     )
     return contract
+
+
+@pytest.fixture(scope="class")
+def event_caller_contract(web3_client, class_account) -> typing.Any:
```
Comment (kristinaNikolaevaa) [inline integration/tests/conftest.py:240]: Are you sure that ./tests/services/test_ws_subscriver.py have access to that conftest.py?

File: integration/tests/basic/test_rpc_calls.py [hunk 4]
```diff
@@ -174,9 +175,55 @@ def test_eth_gas_price(self, param: tp.Union[str, None]):
...
+            assert "error" not in response
+            assert response["result"][0]["blockHash"] == receipt_blockhash
+            assert_fields_are_hex(response["result"][0],
+                                  ["transactionHash", "blockHash",
+                                   "blockNumber", "transactionIndex", "address",
+                                   "logIndex", "data", "transactionLogIndex"])
+            assert_fields_are_boolean(response["result"][0], ["removed"])
+        else:
+            pass
+            # NDEV-2237 bug
```
Comment (kristinaNikolaevaa) [inline integration/tests/basic/test_rpc_calls.py]: maybe better to move it to separeted test? 
It is not good to have tests with if/else (dependences of responce)

File: integration/tests/basic/test_rpc_calls.py [hunk 5]
```diff
@@ -174,9 +175,55 @@ def test_eth_gas_price(self, param: tp.Union[str, None]):
             response["result"]
         ), f"Invalid current gas price `{response['result']}` in wei"
 
+    @pytest.mark.parametrize(
+        "blockhash, tag1, tag2", [
```
Comment (kristinaNikolaevaa) [inline integration/tests/basic/test_rpc_calls.py]: why do you need blockhash as parameter if it is always True

File: integration/tests/basic/test_rpc_calls.py [hunk 6]
```diff
@@ -174,9 +175,55 @@ def test_eth_gas_price(self, param: tp.Union[str, None]):
...
+            (True, None, None),
+            (True, Tag.EARLIEST, None),
+            (True, None, Tag.LATEST),
+        ]
+    )
+    def test_eth_get_logs_blockhash(self, event_caller_contract, blockhash, tag1, tag2):
```
Comment (kristinaNikolaevaa) [inline integration/tests/basic/test_rpc_calls.py]: what about tests with blockhash + address or topic in parameters ? 
And some negative tests for all fields

File: integration/tests/basic/test_rpc_calls.py (hunk 4 above)
Comment (lyubenkov) [reply integration/tests/basic/test_rpc_calls.py]: Done

File: integration/tests/basic/test_rpc_calls.py (hunk 5 above)
Comment (lyubenkov) [reply integration/tests/basic/test_rpc_calls.py]: Removed

File: integration/tests/conftest.py (hunk 3 above)
Comment (lyubenkov) [reply integration/tests/conftest.py:240]: Yes, it's working, as conftest located in the parent directory

File: integration/tests/basic/test_rpc_calls.py (hunk 6 above)
Comment (lyubenkov) [reply integration/tests/basic/test_rpc_calls.py]: Added

**Review by gigimon — APPROVED** [review]

**Comment by gigimon** [comment]
@lyubenkov pls wait for @kristinaNikolaevaa review

File: integration/tests/conftest.py [hunk 7]
```diff
@@ -1,5 +1,6 @@
 import os
 import random
+import typing
```
Comment (kristinaNikolaevaa) [inline integration/tests/conftest.py:3]: import typing twice

**Review by kristinaNikolaevaa — APPROVED** [review]

File: integration/tests/conftest.py (hunk 7 above)
Comment (lyubenkov) [reply integration/tests/conftest.py:3]: Fixed

**Pull request merged**